### PR TITLE
[SC-166] Add generic xchain forwarders library

### PR DESCRIPTION
**Pull request by hexonaut**
Smooth over the complexities of bridging with this unified library.

### CODE REVIEW

**Comment by hexonaut** [comment]
Ready for review.

**Review by barrutko — APPROVED** [review]

File: src/XChainForwarders.sol [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        );
+    }
+
+    function sendMessageOptimismMainnet(
+        address target,
+        bytes memory message,
+        uint256 gasLimit
+    ) internal {
+        sendMessageOptimism(
+            0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1,
```
Comment (lucas-manuel) [inline src/XChainForwarders.sol:54]: Can we make these all constants at the top?

File: src/XChainForwarders.sol [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+ * @title XChainForwarders
+ * @notice Helper functions to abstract over L1 -> L2 message passing.
+ * @dev General structure is sendMessageXXX(target, message, gasLimit) where XXX is the remote domain name (IE OptimismMainnet, ArbitrumOne, Base, etc).
+ */
+library XChainForwarders {
+
+    /// ================================ Optimism Style ================================
+
+    function sendMessageOptimism(
+        address l1CrossDomain,
```
Comment (lucas-manuel) [inline src/XChainForwarders.sol:36]: This name wasn't immediately clear to me, maybe something to indicate its an AMB? `l1CrossDomainAMB`?

**Review by lucas-manuel — CHANGES_REQUESTED** [review]

**Review by lucas-manuel — CHANGES_REQUESTED** [review]
Constants

**Review by lucas-manuel — APPROVED** [review]

**Pull request merged**